src/Stoa.ts

```diff
@@ -147,22 +156,102 @@ class Stoa {
                 return;
             }
 
-            console.log(block);
+            // Use Promise to record block data synchronously.
+            let save = (storage: LedgerStorage, block: Block) =>
+            {
+                return new Promise<boolean>((resolve, reject) => {
+                    storage.putBlocks(block, ()=>{resolve(true);}, (error)=>{reject(error);});
+                });
+            };
+
+            // Extract the block height from JSON.
+            let getBlockHeight = (block: any): number =>
+            {
+                if  (
+                    (block.header !== undefined) &&
+                    (block.header.height !== undefined) &&
+                    (block.header.height.value !== undefined)
+                )
+                {
+                    return Number(block.header.height.value);
+                }
+                else
+                {
+                    throw Error("Not found block height in JSON Block");
+                }
+            };
+
+            try
+            {
+                let height = getBlockHeight(block);
 
-            this.ledger_storage.putBlocks(block,
-                () =>
+                // Obtain the height of the block that should be recorded in the database.
+                let expected_height = await this.ledger_storage.getExpectedBlockHeight();
+                if (height == expected_height)
                 {
+                    // The Normal case
+
+                    // Save the block just received
+                    await save(this.ledger_storage, block);
+                    this.pushSaveLog(`save ${height}`);
```

Comment:
`save 1`
How would you describe it
`Block storage success #1`

Reply:
This will only be used for verification in the test code.
Only the declaration of pushSaveLog, no implementation.
Its implementation is only in the test code.

Reply:
```suggestion
                    this.pushSaveLog(`Block storage success # ${height}`);
```
Yes, we'd better make it into a more specific expression of the log.

Reply:
I solved this IRL.

Reply:
I just added a log output function separately.